src/services/anchor-service.ts

```diff
@@ -145,38 +145,53 @@ export class AnchorService {
       }
     }
 
-    logger.imp(`Creating Merkle tree from ${candidates.length} selected streams`)
-    const merkleTree = await this._buildMerkleTree(candidates)
+    try {
+      logger.imp(`Creating Merkle tree from ${candidates.length} selected streams`)
+      const merkleTree = await this._buildMerkleTree(candidates)
 
-    // create and send ETH transaction
-    logger.debug('Preparing to send transaction to put merkle root on blockchain')
-    const tx: Transaction = await this.blockchainService.sendTransaction(
-      merkleTree.getRoot().data.cid
-    )
+      // create and send ETH transaction
+      logger.debug('Preparing to send transaction to put merkle root on blockchain')
+      const tx: Transaction = await this.blockchainService.sendTransaction(
+        merkleTree.getRoot().data.cid
+      )
 
-    // create proof on IPFS
-    logger.debug('Creating IPFS anchor proof')
-    const ipfsProofCid = await this._createIPFSProof(tx, merkleTree.getRoot().data.cid)
+      // create proof on IPFS
+      logger.debug('Creating IPFS anchor proof')
+      const ipfsProofCid = await this._createIPFSProof(tx, merkleTree.getRoot().data.cid)
 
-    // create anchor records on IPFS
-    logger.debug('Creating anchor commits')
-    const anchors = await this._createAnchorCommits(ipfsProofCid, merkleTree)
+      // create anchor records on IPFS
+      logger.debug('Creating anchor commits')
+      const anchors = await this._createAnchorCommits(ipfsProofCid, merkleTree)
 
-    // Update the database to record the successful anchors
-    logger.debug('Persisting results to local database')
-    const numAnchoredRequests = await this._persistAnchorResult(anchors, candidates)
+      // Update the database to record the successful anchors
+      logger.debug('Persisting results to local database')
+      const numAnchoredRequests = await this._persistAnchorResult(anchors, candidates)
 
-    logger.imp(`Service successfully anchored ${anchors.length} CIDs.`)
-    return {
-      acceptedRequestsCount: groupedRequests.acceptedRequests.length,
-      alreadyAnchoredRequestsCount: groupedRequests.alreadyAnchoredRequests.length,
-      anchoredRequestsCount: numAnchoredRequests,
-      conflictingRequestCount: groupedRequests.conflictingRequests.length,
-      failedRequestsCount: groupedRequests.failedRequests.length,
-      failedToPublishAnchorCommitCount: merkleTree.getLeaves().length - anchors.length,
-      unprocessedRequestCount: groupedRequests.unprocessedRequests.length,
-      candidateCount: candidates.length,
-      anchorCount: anchors.length,
+      logger.imp(`Service successfully anchored ${anchors.length} CIDs.`)
+      return {
```

Comment:
We should still try to log an anchor batch summary message even on failed batches.  So I think what that probably means is moving this return statement out of the try-catch, but making sure that on the error path all the accepted requests get counted as failed.

Reply:
To preserve throwing the errors I instead created a helper function that creates and logs the anchor summary. This helper function is called instead of retuning the anchor summary. 

In regards to "making sure that on the error path all the accepted requests get counted as failed", I have instead added a field called `retryingRequestsCount` for these. I am hesitant to add them to the failed count because `failed` currently means that we have purposeful marked the request as failed. `retryingRequestsCount` means something went wrong. I do not feel strongly tho so if you prefer for me to add them to the failed count I will definitely do so.